dev/release/verify-release-candidate.bat

```diff
@@ -71,7 +71,10 @@ pushd %ARROW_SOURCE%\cpp\build
 @rem This is the path for Visual Studio Community 2017
 call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat" -arch=amd64
 
-cmake -G "Ninja" ^
```

Comment:
If anyone knows how to control the MSVC version used by Ninja let me know

Reply:
This is usually done by calling the `vcvarsall.bat` script for the particular MSVC install (or apparently `VsDevCmd` in more recent versions).

Reply:
We had this conversation in the past, I think. The MSVC "environment setup" scripts provided by VS are more crude than what can be obtained using the `-G` flag. 

Using an explicit generator it's possible to choose the exact toolchain you want, but you must use msbuild, not Ninja. It would be interesting to know if there's a way to both use Ninja and select the target toolchain. The answer may be no but anyway it would be nice to know

Reply:
I didn't try it but we may be able to use `-DMSVC_TOOLSET_VERSION=142` for Visual C++ 2019 or something: https://cmake.org/cmake/help/latest/variable/MSVC_TOOLSET_VERSION.html#variable:MSVC_TOOLSET_VERSION